Packs/AzureNetworkSecurityGroups/Integrations/AzureNetworkSecurityGroups/AzureNetworkSecurityGroups.py

```diff
@@ -239,30 +255,57 @@ def update_rule_command(client: AzureNSGClient, security_group_name: str, securi
 
     rule = client.get_rule(security_group_name, security_rule_name)
     properties = rule.get('properties')
+    properties_keys = list(properties.keys())
 
     updated_properties = assign_params(protocol='*' if protocol == 'Any' else protocol,
-                                       sourceAddressPrefix='*' if source == 'Any' else source,
-                                       destinationAddressPrefix='*' if destination == 'Any' else destination,
                                        access=action, priority=priority,
                                        direction=direction, description=description)
     if source_ports:
         source_ports_list = argToList(source_ports)
         if len(source_ports_list) > 1:
-            properties.pop("sourcePortRange")  # Can't supply both sourcePortRange and sourcePortRanges
+            if "sourcePortRange" in properties_keys:
+                properties.pop("sourcePortRange")  # Can't supply both sourcePortRange and sourcePortRanges
```

Comment:
you can avoid the KeyError exception by passing a default value.
if there is only one range, can the `sourcePortRanges` field be used? this could save the trouble of checking and splitting the code into different cases. if not, keep in mind that you are passing `sourcePortRange` a list value.

relevant for all similar code in the function
```suggestion
            properties.pop("sourcePortRange", None)  # Can't supply both sourcePortRange and sourcePortRanges
```